File: solutions/3012. Minimize Length of Array Using Operations/3012.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
  def minimumArrayLength(self, nums: List[int]) -> int:
    minNum = min(nums)
    
    if any(num % minNum > 0 for num in nums):
      return 1

    return (nums.count(minNum) + 1) // 2

File: solutions/3012. Minimize Length of Array Using Operations/3012.cpp
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
 public:
  int minimumArrayLength(const vector<int>& nums) {
    const int minNum = ranges::min(nums);
    
    if (ranges::any_of(nums, [minNum](int num) {return num % minNum > 0; }))
      return 1;

    return (ranges::count(nums, minNum) + 1) / 2;    
  }
};